ThirdParty/ecec/src/encrypt.c

```diff
@@ -0,0 +1,292 @@
+#include "ece/keys.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <openssl/ec.h>
+#include <openssl/evp.h>
+#include <openssl/rand.h>
+
+// Writes an unsigned 32-bit integer in network byte order.
+static inline void
+ece_write_uint32_be(uint8_t* bytes, uint32_t value) {
+  bytes[0] = (value >> 24) & 0xff;
+  bytes[1] = (value >> 16) & 0xff;
+  bytes[2] = (value >> 8) & 0xff;
+  bytes[3] = value & 0xff;
+}
+
+// Encrypts a plaintext block with optional padding.
+static int
+ece_aes128gcm_encrypt_block(EVP_CIPHER_CTX* ctx, const uint8_t* key,
+                            const uint8_t* iv, const uint8_t* block,
+                            size_t blockLen, const uint8_t* pad, size_t padLen,
+                            uint8_t* record) {
+  int err = ECE_OK;
+
+  // Encrypt the plaintext and padding.
+  if (EVP_EncryptInit_ex(ctx, EVP_aes_128_gcm(), NULL, key, iv) <= 0) {
+    err = ECE_ERROR_ENCRYPT;
+    goto end;
+  }
+  int chunkLen = -1;
+  if (EVP_EncryptUpdate(ctx, record, &chunkLen, block, (int) blockLen) <= 0) {
```

Comment:
`EVP_CIPHER_block_size(EVP_aes_128_gcm()) == 1`, so this seems like it's accurate. I added a comment about this in tip-of-tree ecec. Do you think it's worth `assert`-ing that, or allocating the extra byte so it's visibly safe?